### DISCUSSION TITLE
Add another way to stop the crypto currency token

### DISCUSSION THREAD

**Discussion by Urban-Hacker** [ochadenas/cpudefense#126] [Ideas]
Currently when you are dealing with cryptocurrency token there are only two "towers" that can destroy / weaken them. The SHF and the SUB.

Maybe we could have another option that target only the crypto token, maybe something to reveal its value and turn it into a normal high level attacker.

I think it would be nice and give the player a potential "tradeoff" between destroying some of his network by selling chips or trying to directly hit the cryptotoken like we have now.

I wonder what other thinks about this

**Comment by Kutabu** [comment]
I think the coins are important for the game balance. They prevent you from building a long chain of ACC chips and intercepting the accumulated attackers in MEM at the end. It also ensures that you have to be careful when you secure very rarely travelled paths exclusively with MEM.

But anyway, you're right. The fact that only two chips affect the coins has also bothered me. Especially when the coins are hidden among many attackers. Sometimes the chips only attack the normal attackers and the SHL and SUB don't attack the coin at all.
But since the chips "intelligently" choose the attackers, they could prioritise the coins. I'd have to see how he solved this in the code...

**Reply by Cwpute** [reply]
I'd like to agree with you when you say cryptocoins help balance the game out. But since they're so few of them per-level, and some don't even feature any as attackers (only as rewards), i can't. Maybe if crypto-coins were an entirely different type of attacker, fully aknoledged and integrated, with a different behavior for example, dedicated chips to target them (as suggested by @Urban-Hacker) and dedicated waves of them (which would result in a different value for them, but that's another topic) yeah, maybe then i could agree.
But right now, the game balance they contribute to isn't rich, and i feel its rather a by-product of their design (they can't be swallowed, only destroyed) rather than a weighted decision.

**Comment by ochadenas** [comment]
Well, the fact that the coins are immune to MEM and ACC in fact _is_ an attempt to create balance. Otherwise the combination ACC - ACC - ACC - MEM would be unbeatable. 

And I'm a bit at a loss among the discussions: Is the MEM overpowered? Or are crypto-coins to strong? The game oscillates between these two poles ...

I think that maybe a chip that slows down attackers could make sense. This would also affect the coins. Only that I can't think of a logical chip that has the effect of slowing down electric signals. Perhaps a resistor? That would mean stretching the analogy really, really far. But it's just a game, so that's probably fine.

By the way, are you a native French speaker (or is only your phone set to French)? Do you prefer "coin" or "écu" as translation? "Ecu" seems somewhat old-fashioned, but "coin" normally has a different meaning.

**Reply by The-JMFB** [reply]
>And I'm a bit at a loss among the discussions: Is the MEM overpowered? Or are crypto-coins to strong? The game oscillates between these two poles ...

I agree. Perhaps in normal and turbo modes there is some concern about the balance (I don't really think so), but the further into endless mode I travel, the more I feel like there is a happy balance between the 2.

There is so much attention focused on these, rather than expansion ideas that could further minimize their impact anyway. As a currency, the coins become obsolete eventually. I genuinely enjoyed the fact that in normal and turbo mode, those coins dictate my strategy. Not just during the level, but the overarching strategy to get through those modes. It's impossible to max out every hero before completing them, so you have to really consider the investment strategy you want to utilize for success. Sure, you can reset them and change strategies but I found it far more fun to commit to a strategy and pursue it to the end. To that end, I always wanted to get every possible coin. It was critical to my strategy.

Admittedly, in endless mode the coins are nothing more than a minor distraction once you've maxed out all of your heroes. Still, their presence dictates a need to always have a sub on every path, and that helps to balance the strategy necessary to succeed, I think, because the sub alone is nearly worthless. 

The mem chip, in any of the variations that have been used so far, is absolutely essential for endless mode after the first couple dozen levels. The further into endless mode you go, the more balanced they become. Could they be exploited to make the game easier? Sure. Does it impact you? No. Play the game you want to play it and stop trying to make it more difficult for someone that is perfectly happy to play using a weaker strategy.

I use a maximum of 1 mem per exit (CPU) path. I don't ever even consider using more than that, except for, rarely, very early in a level while I'm still building my frontline defense - but I'm at level 120 and the number and strength of attackers in the opening waves sometimes force this strategy. It's not an "easy" or particularly satisfying strategy when I have to use extra mems because they require constant purging. The more I have, the harder it is to keep them purged AND build new defenses.

If these things make you feel like the game is out of balance, then challenge yourself to balance it with more challenging strategies. Stop expecting the game to try to outsmart you and instead outsmart and challenge yourself.

**Comment by Cwpute** [comment]
Yes i'm a native french speaker, and well, i've never seen "écu" used for cryptocoins anywhere before 😄 but i liked it that way, and it worked well enough.
We either use "crypto" as an undefinite "j'ai de la crypto" but never to quantify specific amounts of it. we mostly prefer to use the coin official name for definites "j'ai 3 bitcoins". could we use "token" ? would that be a sufficiently correct term ? if not, you could use "Coin" capitalized, as to not mistake it for the french word for corner which is written the same way.
Or even the basic translated term that should be "pièce", which would technically be correct… whatever you prefer.

Le 20 mars 2024 22:16:51 GMT+01:00, ochadenas ***@***.***> a écrit :

**Comment by Cwpute** [comment]
To add to the discussion: MEM are indeed overpowered, and it's by design: when you build all your early and mid-game game upon progressive value reduction, with bit-wise and hexadecimal calculus, and then suddenly introduce a chip that negates all of this, because it kills an attacker regardless of value, then of course it's going to rock the boat a little bit.
Even ACC chips that merge two attackers into one account for attacker value as an operation involving both will determine the merged attacker's value, so whether it will be beneficial or not for your line of defense.
We've tried several ideas to accomodate for MEM chips but none worked. Seeing how i talk about it, i'm realising maybe they should disapear from chips available to the player for now, until we find an interesting mechanic or concept to take its place.

Cryptocoins… seemed like a good idea to counter the MEM chip, but seem now to only exist because MEM chips were unbalanced in the first place. But if you fight gameplay exceptions with gameplay exceptions, your beautiful interaction of systems is going to fall appart and evolve into a glorified game of rock-paper-scissors where X counters Y, but is weak to Z which is itself weak against Y… 

Yeah, the more i think about it the more i believe MEM chips should be removed right now, until we find something more interesting.


Le 20 mars 2024 22:16:51 GMT+01:00, ochadenas ***@***.***> a écrit :

**Comment by Kutabu** [comment]
I disagree with @Cwpute. The MEM chips are strong but not so overpowered that they should be removed from the game. The idea discussed in #124 to hide them behind a high level hero should be enough to make the two normal game modes not too easy.

Also, I see little need for another way to fight the Cryptocoins. At the moment they fulfil their purpose. Maybe their strength could scale a bit better in endless mode. But because they are quite rare, I see no urgent need for action.
We could rather talk about a different kind of attacker, as is common in other tower defence games. But that would be a complete change to the game mechanics and is nothing for the current development status (imo).

**Reply by Cwpute** [reply]
It's not so much because they are strong that i want them removed, but because **they negate any attacker value**, something every other chip accounts for. Which is in return what makes them overpowered: **they literally don't play by the rules.**

Then to counter this overpowered chip that doesn't play by the rules, you introduce yet another entity that doesn't play by the rules: cryptocoins ! As i was saying: exceptions upon exceptions will turn the game into some kind of rock-paper-scissors if we're not watchful.

**Comment by Urban-Hacker** [comment]
I think its fine to have enemies that are very different to the usual wave you get and I like their mechanics and they force the player to not just rely on mem chips. I think both complement each other quite well.

I just think it would be interesting to have an unique way to defeat the crypto token. Like some unique chip that would fulfill this purpose but be ignored by other attackers.

I think its always nice to offer to players various way to defeat an enemy and it adds one more type of tower / chip.

**Comment by ochadenas** [comment]
Perhaps not a definitive solution, but in 1.40 there is a new hero, Whit Diffie, that reduces the strength of Coins.
